// react-markdown/webpack.config.js
'use strict';

var path = require('path');
var webpack = require('webpack');
var env = JSON.stringify(process.env.NODE_ENV || 'development');

module.exports = {
    entry: [
        path.join(__dirname, 'src', 'react-markdown.js')
    ],
    output: {
        library: 'reactMarkdown',
        libraryTarget: 'umd',
        path: path.join(__dirname, 'umd'),
        filename: 'react-markdown.js'
    },
    externals: {
        react: {
            root: 'React',
            commonjs: 'react',
            commonjs2: 'react',
            amd: 'react'
        }
    },
    module: {
        loaders: [{
            test: /\.json$/,
            loader: 'json'
        }, {
          test: /\.scss$/,
          exclude: /node_modules/,
          loader: ExtractTextPlugin.extract("style-loader", "css-loader!sass-loader"),
          root: path.resolve('./node/static/sass')
        },
          {
            test: /\.(jpe?g|gif|png)$/,
            loader: 'file-loader?emitFile=false&name=[path][name].[ext]'
          }
        ]
    },
    plugins: [
        new webpack.DefinePlugin({'process.env.NODE_ENV': env}),
        new webpack.NoErrorsPlugin(),
        new webpack.optimize.DedupePlugin(),
        new webpack.optimize.OccurenceOrderPlugin(true),
        new webpack.optimize.UglifyJsPlugin()
    ]
};
